app/models/chargeback_container_image.rb

```diff
@@ -28,9 +28,9 @@ def self.build_results_for_report_ChargebackContainerImage(options)
     raise "must provide option :entity_id and provider_id" if id.nil? && provider_id.nil?
 
     @containers = if provider_id == "all"
-                    Container.all
+                    Container.all.joins(:container_group)
                   elsif id == "all"
-                    Container.where('ems_id = ? or old_ems_id = ?', provider_id, provider_id)
+                    Container.joins(:container_group).where('ems_id = ? or old_ems_id = ?', provider_id, provider_id)
                   else
                     Container.joins(:container_group).where('container_groups.container_project_id = ? or container_groups.old_container_project_id = ?', id, id)
                   end
```

Comment:
not quite sure how to beautify this part :sweat:

Reply:
@zeari 
I don\`t prefer method `joins`  because it will fire another query when you ask for `container_group`

(1) **joins**
```ruby 
cont = Container.joins(:container_group)` # 1 query 
cont.container_group # also 1 query
```
but we can use includes
(2) **includes**
- it will fire 2 queries - 1 one for fetching all `containers` and one for `container_groups` (when you are not using where clause, when using where it is 1 query with left joins)
```ruby 
cont = Container.includes(:container_group)` # 2 queries 
cont.container_group # no query
```
(3) **eager_load  (includes/references)**
- it will fire 1 query with left joins

```ruby
cont = Container.includes(:container_group)` # 1 queries 
cont.container_group # no query
```

but one big query(from 3) is not always faster then 2 queries (from 1)

so I think that best solution is to use `includes`  (2)

@kbrock can you add your opinion here ? thanks!

Reply:
Unless you are in a loop and these 2 queries really meant *double* queries (i.e. in the hundreds or thousands), then I wouldn't micro-optimize 2 vs 1 query...prefer readability in whatever you choose.

Reply:
Also please prefer normal where clauses with hashes over strings with `?`